### ISSUE TITLE
memoryview sign issues in 1.8?

### ISSUE THREAD

**Issue by PinkInk** [micropython/micropython#2039]
Hi, memoryview is giving some odd results on these platforms (not tested on pyboard/unix).  This from wipy firmware downloaded today;

```python
>>> b=bytes([127,128,129])
>>> b[0], b[1], b[2]
(127, 128, 129)
>>> mb=memoryview(b)
>>> mb[0], mb[1], mb[2]
(127, -128, -127)
>>>
```

**Comment by peterhinch** [comment]
Same here on a Pyboard V1.1

**Comment by dpgeorge** [comment]
Fixed by 12dd8df3756d58af2f4d1f54eefd90f1ce1e0dbe.
